sessionctx/variable/session.go

```diff
@@ -1245,7 +1246,25 @@ func (s *SessionVars) SetSystemVar(name string, val string) error {
 		s.MetricSchemaRangeDuration = tidbOptInt64(val, DefTiDBMetricSchemaRangeDuration)
 	case CollationConnection, CollationDatabase, CollationServer:
 		if _, err := collate.GetCollationByName(val); err != nil {
-			return errors.Trace(err)
+			var ok bool
+			var charsetVal string
+			var err2 error
+			if name == CollationConnection {
```

Comment:
Using a `switch`?

Reply:
Double `switch` makes the code complicated.
I think it's fine since there are only three branches. If four, then I will use `switch`